test/fixtures/permission-validation/setup.js

```diff
@@ -0,0 +1,490 @@
+  permissions.S1 = {};
+  permissions.S2 = {};
+
+  const createRolePermissions = async ([role, rolePermissions]) => {
+    const preparedPermissions = [];
+
+    for (const permission of rolePermissions) {
+      for (const subsystem of subsystemNames) {
+        if (
+          (permission.Subsystem && permission.Subsystem !== subsystem) ||
+          (permission.onlyForSubsystem &&
+            permission.onlyForSubsystem !== subsystem)
+        ) {
+          continue;
+        }
+
+        const preparedPermission = {};
+        const permissionName = role + permission.Category;
+
+        if (permission.Subsystem) {
+          preparedPermission.Subsystem = subsystems[subsystem];
+        }
+        if (permission.Catalog) {
+          preparedPermission.Catalog = catalogs[permission.Catalog];
+        }
+
+        preparedPermission.Access = permission.Access;
+        preparedPermission.Role = roles[subsystem][role];
+        preparedPermission.Category = categories[permission.Category];
+
+        preparedPermissions.push([
+          subsystem,
+          permissionName,
+          preparedPermission,
+        ]);
+      }
+    }
+
+    await Promise.all(
+      preparedPermissions.map(
+        async ([subsystem, permissionName, permission]) => {
+          permissions[subsystem][permissionName] = await pgProvider.create(
+            'Permission',
+            permission
+          );
+        }
+      )
+    );
+  };
+
+  await Promise.all(Object.entries(rawPermissions).map(createRolePermissions));
+  return permissions;
+};
+
+const linkRolesAndUsers = async (roles, users) => {
+  const linkSubsystemRolesAndUsers = async (roles, users) =>
+    Promise.all(
+      Object.keys(roles).map(async role =>
+        pgProvider.linkDetails(
+          'SystemUser',
+          'Roles',
+          users[`${role}User`],
+          roles[role]
+        )
+      )
+    );
+
+  await Promise.all(
+    subsystemNames.map(subsystem =>
+      linkSubsystemRolesAndUsers(roles[subsystem], users[subsystem])
+    )
+  );
+};
+
+const linkActionsAndPermissions = async (actions, permissions) => {
+  const links = [
+    [permissions.S1.BlockedPerson, actions.getPersonName],
+    [permissions.S1.AdminPerson, actions.getPersonName],
+    [permissions.S1.AdminDocument, actions.getDocumentName],
+    [permissions.S1.AdminPersonDocument, actions.getPersonDocumentName],
+  ];
+
+  await Promise.all(
+    links.map(async args =>
+      pgProvider.linkDetails('Permission', 'Actions', ...args)
+    )
+  );
+};
+
+const linkApplicationsAndRoles = async (roles, applications) => {
+  const links = [
+    [roles.S1.Admin, applications.PersonApplication],
+    [roles.S2.Admin, applications.DocumentApplication],
+  ];
+
+  await Promise.all(
+    links.map(async args =>
+      pgProvider.linkDetails('Role', 'Applications', ...args)
+    )
+  );
+};
+
+const startJstpServer = async () => {
+  const user = {
+    login: async (connection, userId) =>
```

Comment:
There is no need to create another `Promise` inside the function here:
```suggestion
    login: async (connection, userId) => {
      connection.session.set('userId', userId);
    },
```

Reply:
My fault, just an implicit Promise here. Fixed